test/e2e/fixtures.go

```diff
@@ -150,6 +154,151 @@ func (e *Echo) Deploy(ns, name string) func() {
 	}
 }
 
+// EchoSecure manages the TLS-secured ingress-conformance-echo fixture.
+type EchoSecure struct {
+	client client.Client
+	t      ginkgo.GinkgoTInterface
+}
+
+// Deploy creates the TLS-secured ingress-conformance-echo-tls fixture, specifically
+// the deployment and service, in the given namespace and with the given name, or
+// fails the test if it encounters an error. Namespace is defaulted to "default"
+// and name is defaulted to "ingress-conformance-echo-tls" if not provided. Returns
+// a cleanup function.
+func (e *EchoSecure) Deploy(ns, name string) func() {
+	valOrDefault := func(val, defaultVal string) string {
+		if val != "" {
+			return val
+		}
+		return defaultVal
+	}
+
+	ns = valOrDefault(ns, "default")
+	name = valOrDefault(name, "ingress-conformance-echo-tls")
+
+	deployment := &appsv1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: ns,
+			Name:      name,
+		},
+		Spec: appsv1.DeploymentSpec{
+			Selector: &metav1.LabelSelector{
+				MatchLabels: map[string]string{"app.kubernetes.io/name": name},
+			},
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: map[string]string{"app.kubernetes.io/name": name},
+				},
+				Spec: corev1.PodSpec{
+					Containers: []corev1.Container{
+						{
+							Name:  "conformance-echo",
+							Image: "tsaarni/echoserver:latest",
+							Env: []corev1.EnvVar{
```

Comment:
nit: missing from the original one but might be good to add these as well like the basic echo server https://github.com/projectcontour/contour/blob/eb8ff868c365af1f1ef83a31fcfa488f894253b6/test/e2e/fixtures.go#L86-L93

Reply:
Oh yeah, good call, I did that somewhere, let me find it..